src/features/auth/authSlice.ts

```diff
@@ -0,0 +1,129 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import {
+  getUserFromLocalStorage,
+  addUserToLocalStorage,
+  removeUserFromLocalStorage,
+} from "../../utils/localStorage";
+import { axiosClient } from "../../utils/axios";
+import { AxiosError } from "axios";
+import type { RootState } from "../../app/store";
+
+export interface KnownError {
+  msg: string;
+}
+
+export interface User {
+  user: {
+    userId: string;
+    name: string;
+    email: string;
+  };
+  token: string;
+  refreshToken?: string;
+  error?: KnownError;
+}
+
+export interface RegisterPayload {
+  name: string;
+  email: string;
+  password: string;
+}
+
+export interface LoginPayload {
+  email: string;
+  password: string;
+}
+
+export interface AuthState {
+  isLoading: boolean;
+  user: User | null;
+  isError: boolean;
+}
+
+const initialState: AuthState = {
+  isLoading: false,
+  user: getUserFromLocalStorage(),
+  isError: false,
+};
+
+export const registerUser = createAsyncThunk<
+  User,
+  RegisterPayload,
+  { rejectValue: KnownError }
+>(
+  "auth/signup",
+  async (registerPayload: RegisterPayload, { rejectWithValue }) => {
+    try {
+      const response = await axiosClient.post("/auth/register", registerPayload);
```

Comment:
I‘m no problem with `axiosClient`, but we can do a bit of improvement here.
Create a folder called `services` under `src`, and in this folder, create files named like {model}.ts e.g. user.ts.
Then, configure the URL and parameters in these files.

User.ts
```ts
import { axiosClient: api } from "../../utils/axios";
import { LoginPayload } from "types folder"

export const loginApi = (loginPayload: LoginPayload) => api.post("/auth/login", loginPayload)

...
```